docker-api/src/main/java/com/yahoo/vespa/hosted/dockerapi/DockerImpl.java

```diff
@@ -231,6 +233,31 @@ public void deleteContainer(ContainerName containerName) {
         }
     }
 
+    @Override
+    public void updateContainer(ContainerName containerName, ContainerResources resources) {
+        try {
+            UpdateContainerCmd updateContainerCmd = dockerClient.updateContainerCmd(containerName.asString())
+                    .withCpuShares(resources.cpuShares())
+                    .withMemory(resources.memoryBytes())
+
+                    // Command line argument `--cpus c` is sent over to docker daemon as "NanoCPUs", which is the
+                    // value of `c * 1e9`. This however, is just a shorthand for `--cpu-period p` and `--cpu-quota q`
+                    // where p = 100000 and q = c * 100000.
+                    // See: https://docs.docker.com/config/containers/resource_constraints/#configure-the-default-cfs-scheduler
+                    // --cpus requires API 1.25+ on create and 1.29+ on update
+                    // NanoCPUs is supported in docker-java as of 3.1.0 on create and not at all on update
```

Comment:
Docker CLI provides 3 options for configuring CFS on container create or update: `--cpu-period`, `--cpu-quota` and `--cpus` the first 2 go together, and the last one is just shortcut for `--cpu-period 100000` and `--cpu-quota 100000*c` where `c` is the value of `--cpu`, but for whatever reason, this conversion is not done in the client, but on the server (the daemon): https://github.com/docker/docker-ce/blob/8a582e89b407639018e067123eb17f61dbfd10ee/components/engine/daemon/update_linux.go#L29:L32

`CpuPeriod` and `CpuQuota` was added to `HostConfig` (part of the JSON sent during container create command) in docker 1.7.0 (API version 1.19), while `NanoCPUs` was added in docker 1.13 (1.25).
The update command was added in docker 1.10 (1.22) and originally included `CpuQuota` and `CpuPeriod`. `NanoCPUs` was added to update in docker 17.04 (1.28).
https://docs.docker.com/engine/api/version-history/

And then there is `docker-java`:
We use `docker-java` 3.0.13 which already has `CpuQuota` and `CpuPeriod` both in `HostConfig` and `UpdateContainerCmd`. `NanoCPUs` was added to `HostConfig` in 3.1.0-rc6 (latest is 3.1.0-rc7).
As of today, `NanoCPUs` has still has not been added to `UpdateContainerCmd`.

What I'm doing in this PR is simply emulating `NanoCPUs` in the client using `CpuQuota` and `CpuPeriod` so that when it's added to `docker-java`, we can simplify those 2 setters to 1.

Reply:
Believe it or not, I find this super-useful.

For the comment, add a sentence at the end about your intention? (once NanoCPUs are available, use that instead of quota/period etc etc)